packages/opencensus-core/src/trace/instrumentation/types.ts

```diff
@@ -24,31 +24,45 @@ export interface Plugin {
    * @param moduleExports nodejs module exports from the module to patch
    * @param tracer a tracer instance
    * @param version version of the current instaled module to patch
+   * @param options plugin options
    * @param basedir module absolute path
    */
   enable(
       // tslint:disable-next-line:no-any
       moduleExports: any, tracer: Tracer, version: string,
+      options: PluginConfig,
       // tslint:disable-next-line:no-any
       basedir?: string): any;
   /** Method to disable the instrumentation  */
   disable(): void;
 }
 
+export type PluginConfig = {
+  // tslint:disable-next-line:no-any
+  [key: string]: any;
+};
+
+export type NamedPluginConfig = {
+  module: string;
+}&PluginConfig;
 
 /**
  * Type PluginNames: each key should be the name of the module to trace,
  * and its value should be the name of the package  which has the
  * plugin implementation.
  */
 export type PluginNames = {
-  [pluginName: string]: string;
+  [pluginName: string]: string|PluginConfig;
```

Comment:
Should this be `string|NamedPluginConfig` instead? That will allow typescript to know that `module` needs to be defined in the config object. Right now there's no compiler check that verifies that property exists. (Also see the object structure comment above.)